### ISSUE TITLE
Map support

### ISSUE THREAD

**Issue by Xuanwo** [Xuanwo/serde-go#1]
```go
type IndexBasedSegment struct {
	path string
	id   string

	p map[int]*IndexBasedPart
	l sync.RWMutex
}
```

serde should support  deserialize and serialize map here.

**Comment by Xuanwo** [comment]
https://github.com/Xuanwo/serde-go/blob/d398fc554b63e95f8b818193f34f55718a74ca1b/cmd/serde/main.go#L105-L110

We should handle `StarExpr`, `ArrayExpr`, `InterfaceType` and so on here.

**Comment by Xuanwo** [comment]
Map support implemented in #3
